### PR TITLE
[NL Next] Logic to construct Utterance (3/5)

### PR DESCRIPTION
**Pull request by pradh**
This builds on the [Utterance data structure PR](https://github.com/datacommonsorg/website/pull/2092).

Given classification results and past Utterance history, constructs a new Utterance, which includes a list of chart specs.

Most query types (SIMPLE, CONTAINED-IN, RANKING) follow a common processing function `populate_charts()`, with custom callbacks.   Only CORRELATION uses a different handler.

TODO: add unit tests

### CODE REVIEW

File: server/lib/nl/nl_utils.py [hunk 1]
```diff
@@ -89,4 +92,107 @@ def combine_stop_words() -> Set[str]:
...
+#
+def get_sample_child_places(main_place_dcid: str,
+                            contained_place_type: str) -> List[str]:
+  """Find a sampled child place"""
+  logging.info('_sample_child_place: for %s - %s', main_place_dcid,
+               contained_place_type)
+  if not contained_place_type:
+    return None
+  if contained_place_type == "City":
+    return "geoId/0667000"
```
Comment (beets) [inline server/lib/nl/nl_utils.py:149]: maybe we should pick a "simpler" city (one that's not also a county)

File: server/lib/nl/nl_utils.py [hunk 2]
```diff
@@ -89,4 +92,107 @@ def combine_stop_words() -> Set[str]:
...
+  for sv in svs:
+    for _, exist in sv_existence['variable'][sv]['entity'].items():
+      if not exist:
+        continue
+      existing_svs.add(sv)
+
+  return list(existing_svs)
+
+
+_NUM_CHILD_PLACES_FOR_EXISTENCE = 20
```
Comment (beets) [inline server/lib/nl/nl_utils.py:132]: super nit: all constants at the top of the file?

File: server/lib/nl/nl_utils.py [hunk 3]
```diff
@@ -89,4 +92,107 @@ def combine_stop_words() -> Set[str]:
...
+        if child_places:
+          logging.info(
+              '_sample_child_place returning %s',
+              ', '.join(child_places[:_NUM_CHILD_PLACES_FOR_EXISTENCE]))
+          return child_places[:_NUM_CHILD_PLACES_FOR_EXISTENCE]
+  logging.info('_sample_child_place returning %s', main_place_dcid)
+  return [main_place_dcid]
+
+
+_CHART_TITLE_CONFIG_RELATIVE_PATH = "../../config/nl_page/chart_titles_by_sv.json"
```
Comment (beets) [inline server/lib/nl/nl_utils.py:175]: ditto

**Review by beets — APPROVED** [review]
nice!

nit: more of the helpers / utils / data structures could move into a new module. as followups once these 5 PRs are merged?

File: server/lib/nl/nl_utils.py (hunk 2 above)
Comment (pradh) [reply server/lib/nl/nl_utils.py]: Done.

File: server/lib/nl/nl_utils.py (hunk 1 above)
Comment (pradh) [reply server/lib/nl/nl_utils.py:150]: SG. picked sunnyvale (not a county, nor a common name)

**Review by pradh — COMMENTED** [review]
Thanks!

File: server/lib/nl/nl_utils.py (hunk 3 above)
Comment (pradh) [reply server/lib/nl/nl_utils.py]: Done.

**Comment by pradh** [comment]
> nice!
> 
> nit: more of the helpers / utils / data structures could move into a new module. as followups once these 5 PRs are merged?

Thanks for the review!  Moving more helpers sounds good!

**Pull request merged**